azure_functions_worker/testutils.py

```diff
@@ -239,6 +247,115 @@ def _run_test(self, test, *args, **kwargs):
+
+    def get_new_mem_map_name(self):
+        return str(uuid.uuid4())
+
+    def get_random_bytes(self, num_bytes):
+        return bytearray(random.getrandbits(8) for _ in range(num_bytes))
+
+    def get_random_string(self, num_chars):
+        return ''.join(random.choices(string.ascii_uppercase + string.digits,
+                                      k=num_chars))
+
+    def is_valid_uuid(self, uuid_to_test: str, version: int = 4) -> bool:
+        """
+        Check if uuid_to_test is a valid UUID.
+        Reference: https://stackoverflow.com/a/33245493/3132415
+        """
+        try:
+            uuid_obj = uuid.UUID(uuid_to_test, version=version)
+        except ValueError:
+            return False
+        return str(uuid_obj) == uuid_to_test
+
+    def _createSharedMemoryDirectories(self, directories):
+        for temp_dir in directories:
+            temp_dir_path = os.path.join(temp_dir, consts.UNIX_TEMP_DIR_SUFFIX)
+            if not os.path.exists(temp_dir_path):
+                os.makedirs(temp_dir_path)
+
+    def _deleteSharedMemoryDirectories(self, directories):
+        for temp_dir in directories:
+            temp_dir_path = os.path.join(temp_dir, consts.UNIX_TEMP_DIR_SUFFIX)
+            shutil.rmtree(temp_dir_path)
+
+    def _setUpLinux(self):
+        self._createSharedMemoryDirectories(consts.UNIX_TEMP_DIRS)
+
+    def _tearDownLinux(self):
+        self._deleteSharedMemoryDirectories(consts.UNIX_TEMP_DIRS)
+
+    def _setUpDarwin(self):
+        """
+        Create a RAM disk on macOS.
+        Ref: https://stackoverflow.com/a/2033417/3132415
+        """
+        size_in_mb = consts.MAX_BYTES_FOR_SHARED_MEM_TRANSFER / (1024 * 1024)
+        size = 2048 * size_in_mb
+        # The following command returns the name of the created disk
+        cmd = ['hdutil', 'attach', '-nomount', f'ram://{size}']
+        result = subprocess.run(cmd, stdout=subprocess.PIPE)
+        if result.returncode != 0:
+            raise IOError(f'Cannot create ram disk with command: {cmd} - '
+                          f'{result.stdout} - {result.stderr}')
+        disk_name = result.stdout
+        # We create a volume on the disk created above and mount it
+        volume_name = 'shm'
+        cmd = ['diskutil', 'eraseVolume', 'HFS+', volume_name, disk_name]
+        result = subprocess.run(cmd, stdout=subprocess.PIPE)
+        if result.returncode != 0:
+            raise IOError(f'Cannot create volume with command: {cmd} - '
+                          f'{result.stdout} - {result.stderr}')
+        directory = f'/Volumes/{volume_name}'
+        self.created_directories = [directory]
+        self._createSharedMemoryDirectories(self.created_directories)
+        # Override the AppSetting for the duration of this test so the
+        # FileAccessorUnix can use these directories for creating memory maps
+        os.environ.update(
+            {UNIX_SHARED_MEMORY_DIRECTORIES: self.created_directories})
```

Comment:
You are directly assigning a list. Change this to a comma-sep value

```python
os.environ.update({UNIX_SHARED_MEMORY_DIRECTORIES:
  ",".join(self.created_directories)})
```

```python
―――――――――――――――――――――――――― TestSharedMemoryMap.test_dispose_without_delete_file ―――――――――――――――――――――――――――
Traceback (most recent call last):
  File "/Users/varadmeru/work/microsoft/azfunctions/pyworker/azure_functions_worker/testutils.py", line 269, in setUp
    self._setUpDarwin()
  File "/Users/varadmeru/work/microsoft/azfunctions/pyworker/azure_functions_worker/testutils.py", line 353, in _setUpDarwin
    os.environ.update(
  File "/usr/local/Cellar/python@3.8/3.8.6/Frameworks/Python.framework/Versions/3.8/lib/python3.8/_collections_abc.py", line 832, in update
    self[key] = other[key]
  File "/usr/local/Cellar/python@3.8/3.8.6/Frameworks/Python.framework/Versions/3.8/lib/python3.8/os.py", line 680, in __setitem__
    value = self.encodevalue(value)
  File "/usr/local/Cellar/python@3.8/3.8.6/Frameworks/Python.framework/Versions/3.8/lib/python3.8/os.py", line 750, in encode
    raise TypeError("str expected, not %s" % type(value).__name__)
TypeError: str expected, not list
```

Reply:
With these changes done: here's the result for `TestSharedMemoryMap`

```python
 tests/unittests/test_shared_memory_map.py ⨯✓✓✓✓✓                                           100% ██████████
========================================= short test summary info =========================================
FAILED tests/unittests/test_shared_memory_map.py::TestSharedMemoryMap::test_dispose_without_delete_file

Results (19.03s):
       5 passed
       1 failed
         - tests/unittests/test_shared_memory_map.py:108 TestSharedMemoryMap.test_dispose_without_delete_file
```

Reply:
Thanks a lot for trying this out on your machine!